ranger/ranger.go

```diff
@@ -43,13 +44,22 @@ type Ranger struct {
 	users     middleware.UserStorer
 }
 
-// New constructs a Ranger from the provided options.
+// Config holds the data that is requried to init a new ranger. This also allow us to
+// guarantee that certain attributes have been hyrdrated when running the options.
+type Config struct {
+	Env trails.Environment
```

Comment:
Given this is exposed to users, lets start documenting these fields for users to have clarity about what it means since the functional options had comments explaining them.